plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/conditions/accessors/HardcodedPropertyAccessorRegistry.java

```diff
@@ -0,0 +1,131 @@
+        propertyAccessors.put(Item.class.getName(), new ItemHardcodedPropertyAccessor(this));
+        propertyAccessors.put(MetadataItem.class.getName(), new MetadataItemHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Metadata.class.getName(), new MetadataHardcodedPropertyAccessor(this));
+        propertyAccessors.put(TimestampedItem.class.getName(), new TimestampedItemHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Event.class.getName(), new EventHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Profile.class.getName(), new ProfileHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Consent.class.getName(), new ConsentHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Session.class.getName(), new SessionHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Rule.class.getName(), new RuleHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Goal.class.getName(), new GoalHardcodedPropertyAccessor(this));
+        propertyAccessors.put(CustomItem.class.getName(), new CustomItemHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Campaign.class.getName(), new CampaignHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Map.class.getName(), new MapHardcodedPropertyAccessor(this));
+    }
+
+    public static class NextTokens {
+        public String propertyName;
+        public String leftoverExpression;
+    }
+
+    protected NextTokens getNextTokens(String expression) {
+        NextTokens nextTokens = new NextTokens();
+        if (expression.startsWith("[\"")) {
+            int lookupNameBeginPos = "[\"".length();
+            int lookupNameEndPos = expression.indexOf("\"].", lookupNameBeginPos);
+            if (lookupNameEndPos > lookupNameBeginPos) {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos, lookupNameEndPos);
+                nextTokens.leftoverExpression = expression.substring(lookupNameEndPos+2);
+            } else {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos);
+                nextTokens.leftoverExpression = null;
+            }
+        } else if (expression.startsWith(".")) {
+            int lookupNameBeginPos = ".".length();
+            int lookupNameEndPos = lookupNameBeginPos;
+            int dotlookupNameEndPos = expression.indexOf(".", lookupNameBeginPos);
+            int squareBracketlookupNameEndPos = expression.indexOf("[", lookupNameBeginPos);
+            if (dotlookupNameEndPos > lookupNameBeginPos && squareBracketlookupNameEndPos > lookupNameBeginPos) {
+                lookupNameEndPos = Math.min(dotlookupNameEndPos, squareBracketlookupNameEndPos);
+            } else if (dotlookupNameEndPos > lookupNameBeginPos) {
+                lookupNameEndPos = dotlookupNameEndPos;
+            } else if (squareBracketlookupNameEndPos > lookupNameBeginPos) {
+                lookupNameEndPos = squareBracketlookupNameEndPos;
```

Comment:
the two else if seem's unecessary if we do:
`dotlookupNameEndPos > lookupNameBeginPos || squareBracketlookupNameEndPos > lookupNameBeginPos`
in the first condition